### PR TITLE
Output Github job summary

### PR DESCRIPTION
**Pull request by schneems**
From Colin: https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/

Renamed `bin/print_gem_version` to `bin/print_summary`. It now also prints out the Ruby version and is formatted to be markdown.

### CODE REVIEW

File: .github/workflows/build_ruby.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
       - name: Build and package Ruby runtime
         run: bin/build_ruby "$STACK" "${{inputs.ruby_version}}"
       - name: Verify ruby executable and output rubygems version
-        run: bin/print_gem_version "$STACK" "${{inputs.ruby_version}}"
+        run: bin/print_summary "$STACK" "${{inputs.ruby_version}}" >> $GITHUB_STEP_SUMMARY
```
Comment (schneems) [inline .github/workflows/build_ruby.yml:41]: It's not showing for some reason :/

https://github.com/heroku/docker-heroku-ruby-builder/actions/runs/8527371617/job/23358666172

image

Comment (schneems) [reply .github/workflows/build_ruby.yml:41]: It also hides the contents. We should probably `tee` or make sure it shows in the output even if the summary feature doesn't work.

Comment (schneems) [reply .github/workflows/build_ruby.yml]: It is available on the summary page, not the individual action page

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -47,7 +47,7 @@ jobs:
       - name: Build and package Ruby runtime
         run: bin/build_ruby "${{matrix.stack}}" "${{matrix.version}}"
       - name: Verify ruby executable and output rubygems version
-        run: bin/print_gem_version "${{matrix.stack}}" "${{matrix.version}}"
+        run: bin/print_summary "${{matrix.stack}}" "${{matrix.version}}" >> $GITHUB_STEP_SUMMARY
```
Comment (colincasey) [inline .github/workflows/ci.yml:50]: So not using `tee` for the CI job?

**Review by colincasey — APPROVED** [review]

**Pull request merged**